app/assets/javascripts/location.coffee

```diff
@@ -0,0 +1,84 @@
+$('.address_auto_complete_field, .edit_address_auto_complete_field').livequery ->
+  window.placeAutocomplete = new google.maps.places.Autocomplete($(this)[0], types: [ 'geocode' ])
+  placeAutocomplete.addListener('place_changed', fillInAddress);
+
+componentForm =
+  location_administrative_area_level_2: 'short_name'
+  location_locality: 'long_name'
+  location_administrative_area_level_1: 'short_name'
+  location_country: 'long_name'
+  location_sublocality_level_1: 'long_name'
+  location_postal_code: 'short_name'
+
+fillInAddress = ->
+  place = placeAutocomplete.getPlace()
+  $.each componentForm, (index, value) ->
+    document.getElementById(index).value = ''
+    document.getElementById(index).disabled = false
+  if place.address_components.length > 1
+    $.each place.address_components, (component, field) ->
+      addressType = field.types[0]
+      AddressType = "location_"+addressType
+      if componentForm[AddressType]
+        val = field.long_name
+        if $('#'+AddressType).length > 0
```

Comment:
@deepali1-aconnexion can you please add spaces around `+` symbol? I think it will be more readable.